finality-aleph/src/network.rs

```diff
@@ -361,6 +370,38 @@ impl<D: Clone + Encode + Decode, B: BlockT + 'static, N: Network<B> + Clone>
             .send_message(*peer_id, self.protocol.clone(), message.encode());
     }
 
+    fn authenticate_to(&self, session_data: &SessionData<D>, peer_id: PeerId) {
+        self.commands_for_session
+            .unbounded_send(SessionCommand::Meta(
+                MetaMessage::Authentication(
+                    session_data.auth_data.clone(),
+                    session_data.auth_signature.clone(),
+                ),
+                Some(peer_id),
+            ))
+            .expect("Sending commands to session should work.");
+    }
+
+    fn on_incoming_meta(&self, message: MetaMessage, peer_id: PeerId) {
+        use MetaMessage::*;
+        match message {
+            Authentication(auth_data, signature) => {
+                // Avoids peers claiming other peers represent their node, which could lead to a
+                // DDoS.
+                if peer_id == auth_data.peer_id {
+                    self.on_incoming_authentication(auth_data, signature);
+                }
```

Comment:
lowly nitpick: we could add `else` branch with a log about that one peer tried to claim some others peers.